cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/CassandraDialect.java

```diff
@@ -183,14 +200,23 @@ public void insertOrUpdateTuple(EntityKey key, Tuple tuple, TupleContext tupleCo
 
 			// insert and update are both 'upsert' in cassandra.
 			Insert insert = insertInto( quote( key.getTable() ) );
-			Object[] columnValues = new Object[updateOps.size()];
+			List<Object> columnValues = new LinkedList<>();
+			Set<String> seenColNames = new HashSet<>();
 			for ( int i = 0; i < updateOps.size(); i++ ) {
 				TupleOperation op = updateOps.get( i );
 				insert.value( quote( op.getColumn() ), QueryBuilder.bindMarker() );
-				columnValues[i] = op.getValue();
+				columnValues.add( op.getValue() );
+				seenColNames.add( op.getColumn() );
+			}
+			for ( int j = 0; j < key.getColumnNames().length; j++ ) {
```

Comment:
I was on Cassandra 2.0; Now I have updated to 2.1 and the errors are gone. It's a bit weird because everything looks right from the log statements. I am fine with requiring 2.1, we only should mention the required version in the docs.